src/etc/pre-commit.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/env bash
+#
+# Call `tidy --bless` before each commit
+# Copy this scripts to .git/hooks to activate,
+# and remove it from .git/hooks to deactivate.
+#
+# For help running bash scripts on Windows,
+# see https://stackoverflow.com/a/6413405/6894799
+#
+
+set -Eeuo pipefail
+
+ROOT_DIR="$(git rev-parse --show-toplevel)";
+COMMAND="$ROOT_DIR/x.py test tidy --bless";
```

Comment:
Unfortunately running `x.py` from a different directory will use a different build cache (https://github.com/rust-lang/rust/pull/71994#issue-414915103). Can you also change to the top-level directory so this doesn't rebuild bootstrap?
```suggestion
ROOT_DIR="$(git rev-parse --show-toplevel)";
COMMAND="$ROOT_DIR/x.py test tidy --bless";
cd "$ROOT_DIR"
```